openlabcmd/openlabcmd/plugins/base.py

```diff
@@ -91,7 +93,12 @@ def recover(self):
         print("Recover:")
         for r_code in self.reasons:
             if r_code in RECOVER_MAPS:
-                recover_cmd = RECOVER_MAPS[r_code]['recover'] % self.cloud
+                if (r_code in constants.SPECIAL_CASES and
+                        self.special_case_codes):
+                    recover_cmd = self.special_process(
```

Comment:
The `special_process` looks like a workaround, but not the final solution, this process is trying to solve the arg format of recover cmd, can we add the `arguments` to make the thing general?

    Recover.ROUTER_EXTERNAL_GW: {
        "recover": "openstack --os-cloud {} openstack router set openlab-router --external-gateway {} ",
        "reason": "- Router external gateway: openlab-subnet doesn't attach "
                  "on openlab-router.",
        "args": ["1", "2", "3"]
    }

When we need to format the recover, we can do something like
```python
"openstack --os-cloud {} openstack router set openlab-router --external-gateway {}".format(*["1", "2", "3"])
```

See more https://docs.python.org/3.6/library/string.html#format-examples